### ISSUE TITLE
Feature suggestion/question: tiled snap zones

### ISSUE THREAD

**Issue by oxc** [esjeon/krohnkite#77]
I have a question/proposal for a new layout type, that allows multiple windows to reside in the same tile.
I'm aware that this might seem kind of contrary to what a tiling window manager is supposed to do, but there are already other layouts that arrange windows in overlapping spaces.

Having all windows spread across the screen does not work very well for my workflow, and I've not been able to adjust it to fully dynamic tiling.
Ideally, I would have something like tiled "zones", in which I position windows by dragging them with the mouse. Multiple windows can reside in the same zone, and are activated by the usual means (tab bar, task switcher, keyboard shortcuts, etc.)

Is this something that can be achieved with krohnkite and (mainly) a new layout, or does it contradict too many assumptions implemented in other places (driver doing the window swapping when dragging, etc.)?

**Comment by esjeon** [comment]
Thanks for the suggestion. Your idea sounds interesting.

Actually, dwm guys already have a layout called "[deck](https://dwm.suckless.org/patches/deck/)", where windows in the "stack" share a single "zone", but the layout is pretty much static (master & stack, just like Tile layout).

It also sounds similar to "[tabbed windows](https://forum.kde.org/viewtopic.php?f=18&t=87977)" feature, which was in KWin. Unlike browser tabs, these "tabs" are managed by KWin. This should be exactly what you want here, only if KWin still supported it.

I don't think implementing this would be that much difficult, but the real problem is always the workflow. Maybe this can be an alternate mode for window dragging, but this likely won't work for many people.

**Comment by esjeon** [comment]
Ouch mistake.

**Comment by oxc** [comment]
Interesting thought, so instead of a separate layout, it could just be a different drag action "Create/Add to Deck" instead of "Swap Windows".

I remember the tabbed windows, but they were only for windows of a single application, correct?

**Comment by esjeon** [comment]
> they were only for windows of a single application, correct?

No, it was a KWin feature and was application-agnostic. You could tab different applications into one window ([example](https://www.youtube.com/watch?v=Ek8oA-UXVEQ)). It's sad that it no longer works.

**Comment by paul** [comment]
i3 has this as well, they call it "stacked" or "tabbed" https://i3wm.org/docs/userguide.html#_changing_the_container_layout
